src/Uno.Toolkit.UI/Behaviors/ItemsRepeaterExtensions.cs

```diff
@@ -324,10 +346,35 @@ private static void OnItemsRepeaterElementPrepared(ItemsRepeater sender, Microso
 
 		SetItemSelection(sender, args.Element, selected);
 	}
+
 	private static void OnItemsRepeaterItemsSourceChanged(DependencyObject sender, DependencyProperty dp)
 	{
 		// When we reach here, ItemsSourceView is already updated.
 		if (sender is not ItemsRepeater ir) return;
+
+		SynchronizeDefaultSelection(ir);
+
+#if HAS_UNO_WINUI || __WASM__
```

Comment:
HAS_UNO_WINUI || __WASM__
are redundant... HAS_UNO_WINUI comprise __WASM__ already
further more __WASM__ doesnt even work here

whats is this branching for?
- just skia? then you need to use RuntimeInformation (see: https://github.com/unoplatform/uno.toolkit.ui/blob/6.1.0/src/Uno.Toolkit.UI/Controls/ExtendedSplashScreen/ExtendedSplashScreen.Windows.crossruntime.cs#L47-L50)
- or, skia OR windows? you need a mixture of above with either `!HAS_UNO` or `WINDOWS`

Reply:
I retested to be sure, so yeah this specific issue is only happening for Wasm, but you're right **\_\_WASM\_\_** doesn't even work here. That's why I ended up adding the **HAS_UNO_WINUI** directive. 

The thing is, on Windows this issue is also present ([not just for ItemsRepeater](https://github.com/unoplatform/uno.extensions/pull/2457/files)). Should we just keep HAS_UNO_WINUI and remove \_\_WASM\_\_ in this PR?

Reply:
Thanks I didn't click on the link you provided because I thought it was just for skia, I should've clicked anyway...